### PR TITLE
Improve debugging for resolver

### PR DESCRIPTION
**Pull request by thaJeztah**
This patch improves debugging for the resolver;

- prefix debug messages with `[resolver]` for easier finding in the daemon logs
- use `A` / `AAAA` for query-types in the logs instead of their numeric code
- add debug messages if the external DNS did not return a result
- print sucessful results (t.b.d.)

@fcrisciani @sanimej I added these messages while debugging something; thought they may be useful in general :smile:

### CODE REVIEW

File: resolver.go [hunk 1]
```diff
@@ -467,22 +469,32 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
...
 			r.forwardQueryEnd()
 			if resp != nil {
+				answers := 0
 				for _, rr := range resp.Answer {
 					h := rr.Header()
 					switch h.Rrtype {
 					case dns.TypeA:
+						answers++
 						ip := rr.(*dns.A).A
+						logrus.Debugf("[resolver] received A record %q for %q from %s:%s", ip, h.Name, proto, extDNS.IPStr)
```
Comment (thaJeztah) [inline resolver.go]: Perhaps these are too much, happy to remove if you think so 👍

Comment (fcrisciani) [inline resolver.go]: as debug I think they can be fine

Comment (sanimej) [inline resolver.go]: @thaJeztah For internal queries we do show the response in the debug mode. Its ok to do it for external queries as well.

**Comment by sanimej** [comment]
LGTM

**Pull request merged**